solidity/utils/test_deployment_oracle_v1_pool.js

```diff
@@ -0,0 +1,173 @@
+  const account = web3.eth.accounts.privateKeyToAccount(PRIVATE_KEY);
+  const web3Func = (func, ...args) => func(web3, account, gasPrice, ...args);
+
+  const addresses = { ETH: Web3.utils.toChecksumAddress("0x".padEnd(42, "e")) };
+  const tokenDecimals = { ETH: 18 };
+
+  /*
+    if (!getConfig().mocMedianizer) {
+        throw new Error("MoC Medinizer contract address is undefined");
+    }*/
+
+  let phase = 0;
+  if (getConfig().phase === undefined) {
+    setConfig({ phase });
+  }
+  const execute = async (transaction, ...args) => {
+    if (getConfig().phase === phase++) {
+      await web3Func(send, transaction, ...args);
+      console.log(`phase ${phase} executed`);
+      setConfig({ phase });
+    }
+  };
+
+  // main contracts
+  const contractRegistry = await web3Func(deploy, "contractRegistry", "ContractRegistry", []);
+  const converterFactory = await web3Func(deploy, "converterFactory", "ConverterFactory", []);
+  const sovrynSwapFormula = await web3Func(deploy, "sovrynSwapFormula", "SovrynSwapFormula", []);
+  const sovrynSwapNetwork = await web3Func(deploy, "sovrynSwapNetwork", "SovrynSwapNetwork", [contractRegistry._address]);
+  const converterUpgrader = await web3Func(deploy, "converterUpgrader", "ConverterUpgrader", [contractRegistry._address, addresses.ETH]);
+  const converterRegistry = await web3Func(deploy, "converterRegistry", "ConverterRegistry", [contractRegistry._address]);
+  const liquidityPoolV1ConverterFactory = await web3Func(deploy, "liquidityPoolV1ConverterFactory", "LiquidityPoolV1ConverterFactory", []);
+  const smartToken = await web3Func(deploy, "smartToken", "SmartToken", ["TOKEN", "TKN", 1]);
+  const oracle = await web3Func(deploy, "oracle", "Oracle", []);
+  const liquidityPoolV1Converter = await web3Func(deploy, "liquidityPoolV1Converter", "LiquidityPoolV1Converter", [smartToken._address, contractRegistry._address, MAX_CONVERSION_FEE]);
+
+  //init sovryn swap formula
+  await execute(sovrynSwapFormula.methods.init());
+
+  // initialize contract registry
+  await execute(contractRegistry.methods.registerAddress(Web3.utils.asciiToHex("ContractRegistry"), contractRegistry._address));
+  await execute(contractRegistry.methods.registerAddress(Web3.utils.asciiToHex("ConverterFactory"), converterFactory._address));
+  await execute(contractRegistry.methods.registerAddress(Web3.utils.asciiToHex("SovrynSwapFormula"), sovrynSwapFormula._address));
+  await execute(contractRegistry.methods.registerAddress(Web3.utils.asciiToHex("SovrynSwapNetwork"), sovrynSwapNetwork._address));
+  await execute(contractRegistry.methods.registerAddress(Web3.utils.asciiToHex("SovrynSwapConverterUpgrader"), converterUpgrader._address));
+  await execute(contractRegistry.methods.registerAddress(Web3.utils.asciiToHex("SovrynSwapConverterRegistry"), converterRegistry._address));
+
+  // initialize converter factory
+  await execute(converterFactory.methods.registerTypedConverterFactory(liquidityPoolV1ConverterFactory._address));
```

Comment:
why?

Reply:
This script was created to deploy the Liquidity converted v1 pool with Oracle. Have I done something incorrect or missed anything?

Reply:
Do you mean this script not needed and I should add the oracle deployment to the test_deployment_rsk.js script?

Reply:
i mean why would you register all of these addresses again although they already have been registered on deployment?

Reply:
Oh okay, removing that part.